### ISSUE TITLE
Nightly not starting on Android 4.3

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#8135]
I do have an Android 4.3 device here, on which an old nightly was working (did not record the version though). I just updated it to the newest nightly and now it fails to start ("Unfortunately c:geo has stopped").

Will try to find debug logs. Can anyone try with an emulator?

**Comment by Lineflyer** [comment]
Log:
```
20:44:01.269 Error AndroidRuntime 6389  FATAL EXCEPTION: network--1
20:44:01.269 Error AndroidRuntime 6389  java.lang.NoClassDefFoundError: java.util.Objects
20:44:01.269 Error AndroidRuntime 6389  	at com.fasterxml.jackson.databind.DeserializationContext.<init>(DeserializationContext.java:158)
20:44:01.269 Error AndroidRuntime 6389  	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.<init>(DefaultDeserializationContext.java:44)
20:44:01.269 Error AndroidRuntime 6389  	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext$Impl.<init>(DefaultDeserializationContext.java:318)
20:44:01.269 Error AndroidRuntime 6389  	at com.fasterxml.jackson.databind.ObjectMapper.<init>(ObjectMapper.java:628)
20:44:01.269 Error AndroidRuntime 6389  	at com.fasterxml.jackson.databind.ObjectMapper.<init>(ObjectMapper.java:531)
20:44:01.269 Error AndroidRuntime 6389  	at cgeo.geocaching.network.Network.<clinit>(Network.java:71)
20:44:01.269 Error AndroidRuntime 6389  	at cgeo.geocaching.network.StatusUpdater.lambda$static$2(StatusUpdater.java:95)
20:44:01.269 Error AndroidRuntime 6389  	at cgeo.geocaching.network.-$$Lambda$StatusUpdater$hlY4uQr8W1FcpUKGIX8-VGNuMbg.run(lambda)
20:44:01.269 Error AndroidRuntime 6389  	at io.reactivex.Scheduler$PeriodicDirectTask.run(Scheduler.java:532)
20:44:01.269 Error AndroidRuntime 6389  	at io.reactivex.Scheduler$Worker$PeriodicTask.run(Scheduler.java:479)
20:44:01.269 Error AndroidRuntime 6389  	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:288)
20:44:01.269 Error AndroidRuntime 6389  	at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:253)
20:44:01.269 Error AndroidRuntime 6389  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
20:44:01.269 Error AndroidRuntime 6389  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
20:44:01.269 Error AndroidRuntime 6389  	at java.lang.Thread.run(Thread.java:841)
20:44:01.269 Warning ActivityManager 833    Force finishing activity cgeo.geocaching/.MainActivity
```

**Comment by bekuno** [comment]
Maybe this bug:
https://github.com/FasterXML/jackson-databind/issues/2599

Fixed for the upcomming jackson 2.10.3

Let's downgrade back to the latest (2.9.10.x). 
I will prepare a PR tomorrow targetting release.

**Comment by moving-bits** [comment]
Can reproduce in emulator with API 17. If I'm not mistaken this comes from the com.fasterxml.jackson version we currently use (2.10.2), which is compatible from API level 19 (Android 4.4) onward only. Last version compatible to lower API levels was 2.7 (see https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.8).

@bekuno 
Can you cross check and test what side effects a downgrade would have?

**Comment by bekuno** [comment]
Yackson 2.8.0 was implemented on revision 0f70155e0e7af432dff73e2fa67bc3a566b92db4 at 09.07.16

The switch to 2.10(.2) was rev c9fb39374a0f6bb3f79186433cc296dfff76bebc at 02.02.20.
Back to this point should no problem.

**Comment by Lineflyer** [comment]
We currently have approx. 7300 users running Android 4.1 - 4.3 out of 800.000 active installations.
We could also decide to cut off a new legacy version after next release.

This would mean:
- Revert to lower library on `release` only
- Set close out warning to <4.4 on `release`
- Make RC and finally a release
- Merge `release` to `legacy`
- For a new feature release afterwards we would then need to release two versions (legacy and release)

**Comment by bekuno** [comment]
We should do it later this year as proposed in #8120

**Comment by moving-bits** [comment]
> Yackson 2.8.0 was implemented on revision [0f70155](https://github.com/cgeo/cgeo/commit/0f70155e0e7af432dff73e2fa67bc3a566b92db4) at 09.07.16
> 
> The switch to 2.10(.2) was rev [c9fb393](https://github.com/cgeo/cgeo/commit/c9fb39374a0f6bb3f79186433cc296dfff76bebc) at 02.02.20.
> Back to this point should no problem.

Ah, so the deprecated API level < 19 with 2.8, but did not actually introduce any incompatibility until 2.10.2? So we should go back to the last version before that breaking change for now, until an update to jackson 2.10 line fixes it for now.

After that, we need to see whether they introduce another incompatibility before we EOL API < 21 (#8120), and stay with the last working version for the time in between.
